dspace-oai/src/main/java/org/dspace/xoai/app/XOAI.java

```diff
@@ -455,7 +458,14 @@ private SolrInputDocument index(Item item)
 
         ByteArrayOutputStream out = new ByteArrayOutputStream();
         XmlOutputContext xmlContext = XmlOutputContext.emptyContext(out, Second);
-        retrieveMetadata(context, item).write(xmlContext);
+        Metadata metadata = retrieveMetadata(context, item);
+
+        // Do any additional metadata element, depends on the plugins
+        for (XOAIItemCompilePlugin xOAIItemCompilePlugin : getxOAIItemCompilePlugins()) {
+            metadata = xOAIItemCompilePlugin.additionalMetadata(context, metadata, item);
```

Comment:
The camel for XOAI abreviation looks weird here `xOAIItemCompilePlugin` I would prefer to have `xoaiItemCompilePlugin`. 
But as referred I'm also not in love with this name since It's not clear to me what it is. Could you call it xoaiExtensionPlugin  or xoaiAdditionalElementPlugin. I have some difficulty in proposing better names. I would need to better understand what the interface helps doing.

Reply:
done